pyproject.toml

```diff
@@ -79,6 +80,23 @@ classifiers = [
 ]
 
 [project.optional-dependencies]
+build = [
+    # keep in sync with [build-system] -> requires, see comments there
+    "meson-python>=0.15.0",
+    "Cython>=3.0.4",
+    "pybind11>=2.10.4",
+    "pythran>=0.14.0",
+    "numpy==1.22.4; python_version<='3.10' and platform_python_implementation != 'PyPy'",
+    "numpy==1.23.2; python_version=='3.11' and platform_python_implementation != 'PyPy'",
+    "numpy>=1.26.0,<1.27; python_version=='3.12'",
+    "numpy>=1.25.0; python_version>='3.9' and platform_python_implementation=='PyPy'",
+    "numpy>=2.0.0.dev0; python_version>='3.13'",
+    # used by `python dev.py build`
+    "rich-click",
+    "doit>=0.36.0",
+    "pydevtool",
+    "ninja",
+]
```

Comment:
maybe these can be included in `[build-system] -> requires` (I assume they're not there for a reason?).